Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -146,12 +146,12 @@ public class ChatChannelController: DataController, DelegateCallable, DataStoreP
 
     /// A boolean value indicating if it should send typing events.
     /// It is `true` if the channel typing events are enabled as well as the user privacy settings.
-    internal var shouldSendTypingEvents: Bool {
-        /// Ignore if user typing indicators privacy settings are disabled. By default, they are enabled.
-        let currentUserPrivacySettings = client.sharedCurrentUserController.currentUser?.privacySettings
-        let isTypingIndicatorsForCurrentUserEnabled = currentUserPrivacySettings?.typingIndicators?.enabled ?? true
-        let isChannelTypingEventsEnabled = channel?.canSendTypingEvents ?? true
-        return isTypingIndicatorsForCurrentUserEnabled && isChannelTypingEventsEnabled
+    func shouldSendTypingEvents(completion: @escaping (Bool) -> Void) {
+        guard channel?.canSendTypingEvents ?? true else {
+            completion(false)
+            return
+        }
+        eventSender.currentUserTypingEventsState(completion: completion)
```

Comment:
This new logic also does not seem correct. The privacy settings should be evaluate first, like it was before 🤔

Reply:
Old implementation:
```
let currentUserPrivacySettings = client.sharedCurrentUserController.currentUser?.privacySettings
        let isTypingIndicatorsForCurrentUserEnabled = currentUserPrivacySettings?.typingIndicators?.enabled ?? true
        let isChannelTypingEventsEnabled = channel?.canSendTypingEvents ?? true
        return isTypingIndicatorsForCurrentUserEnabled && isChannelTypingEventsEnabled
```
It can be true only when both privacy settings and channel capabilities are available, therefore, the functionality should be the same. 🤔